### ISSUE TITLE
[6.0.0beta] undefined symbols (PHP 8.4)

### ISSUE THREAD

**Issue by remicollet** [swoole/swoole-src#5538]
```
$ /opt/remi/php84/root/usr/bin/php --no-php-ini -d extension=curl -d extension=mysqlnd -d extension=pdo --define extension=/dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so --modules

Warning: PHP Startup: Unable to load dynamic library '/dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so' (tried: /dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so (/dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so: undefined symbol: zif_exit), /opt/remi/php84/root/usr/lib64/php/modules//dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so.so (/opt/remi/php84/root/usr/lib64/php/modules//dev/shm/BUILDROOT/php84-php-pecl-swoole6-6.0.0-2.fc39.remi.x86_64/opt/remi/php84/root/usr/lib64/php/modules/swoole.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0

```

**Comment by remicollet** [comment]
Also

`PHP Warning:  PHP Startup: Unable to load dynamic library 'modules/swoole.so' (tried: modules/swoole.so (modules/swoole.so: undefined symbol: pdo_pgsql_scanner), /opt/remi/php84/root/usr/lib64/php/modules/modules/swoole.so.so (/opt/remi/php84/root/usr/lib64/php/modules/modules/swoole.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
`

pdo_pgsql_scanner is not a public API...

**Comment by remicollet** [comment]
Some are fixed in pr #5537 (pdo_pgsql_scanner, pdo_sqlite_scanner)

But zif_exit is not a public API (no PHPAPI attribute)

**Comment by remicollet** [comment]
zif_exit also fixed in PR #5537 (3rd commit)

Perhaps you should ask upstream to make it a public API ?

**Comment by matyhtf** [comment]
fixed
